src/Silverback.Integration.Kafka/Messaging/Broker/KafkaProducer.cs

```diff
@@ -95,7 +100,27 @@ private byte[] GetPartitioningKey(IEnumerable<MessageHeader> headers)
         {
             _logger.LogDebug("Creating Confluent.Kafka.Producer...");
 
-            return new Confluent.Kafka.ProducerBuilder<byte[], byte[]>(Endpoint.Configuration.ConfluentConfig).Build();
+            return new Confluent.Kafka.ProducerBuilder<byte[], byte[]>(Endpoint.Configuration.ConfluentConfig)
```

Comment:
I expect the producer behave the same as the consumer. Currently the statistics and the errors aren't published. I copied the code from the consumer.

Reply:
I'm unsure about the usefulness of this. I may be wrong but I believe that the producer would throw an exception when producing a message if there is any issue so you are tracking and handling all casing already. For the statistics it may make sense though (I still don't see much usage for the statistics, tho).

Anyway, it makes sense for completeness. It would maybe be useful to differentiate the events between `KafkaProducerError` and `KafkaConsumerError`.

By the way, what do you do exactly in your application with those event handlers?